src/sfs_atmos_monthly.fd/sfs_atmos_monthly.f90

```diff
@@ -0,0 +1,304 @@
+program sfs_atmos_monthly
+! - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -+ + + ++
+!$$$  MAIN PROGRAM DOCUMENTATION BLOCK
+!                .      .    .                                       .
+! MAIN PROGRAM:  sfs_atmos_monthly    calculate monthly mean.
+!   PRGMMR: K. ASMAR        ORG: NCEP/EMC     DATE: 05-08-2025
+!
+! ABSTRACT: This program takes as input ~31 grib2 files of SFS
+!           post-processed atmospheric products at 24-hrly
+!           forecasts and calculates the monthly mean.
+!
+! PROGRAM HISTORY LOG:
+!   05-08-25  K. Asmar      Origination and implement at NCEP.
+!
+! USAGE:
+!**   Interface.
+!     ----------
+!       The program runs independently and must run after the
+!       completion of all individual POST runs 
+!
+!     Input Files: 
+!     ------------
+!     master grib2: Post-processed atmospheric products at 24 hour intervals
+!
+!     Output File: 
+!     -------------
+!     outfile: grib2 file containing monthly mean for atmospheric products
+!
+!     Method.
+!     -------
+!       Monthly mean. Mean per gridpoint at each message across 31 days.
+!
+!     Programs called.
+!     ----------
+!       baopenwt
+! 	baopenr
+!       skgb
+!       baread
+!       gb_info
+!       gf_getfld
+!	getgb2
+!       gf_free
+!	gribcreate
+!	addgrid
+!	addfield
+!	gribend
+!	wryte
+!	baclose
+!
+! ATTRIBUTES:
+!   LANGUAGE: FORTRAN 90
+!
+! - - - - - - - - - - - - - 
+  use grib_mod
+  use params
+
+  implicit none 
+
+!*********************************************************************
+  integer :: ierr,k,n,m,i,lu
+  integer :: lengrib_in,lengrib_out
+  integer :: iseek,lskip,lgrib,currlen=0
+  integer :: numfields,numlocal,maxlocal
+  integer :: nfiles,npts,icount,fcount
+  integer(4) :: max_bytes,igds(5)
+  integer :: yy,mm,dd,fhr,factor
+  integer :: listsec0(3),listsec1(13)
+  integer, parameter :: mseek=32000
+  real :: spval=9.99e20  
+  integer :: ipdtmpl(29),idrstmpl(18)
+  integer,dimension(200) :: jids,jpdt,jgdt
+  integer :: jpdtn,jgdtn
+  integer :: ibin_scl,idec_scl,inumbits
+  real(8),allocatable :: datavalues(:,:),means(:,:),&
+  			dummy(:),fout(:)
+  type(gribfield) :: gfld,gfld2
+  character(len=1),allocatable,dimension(:) :: cgrib_in,cgrib_out
+  character(len=255),dimension(50) :: ingrib   
+  character*255 :: outfile
+  namelist /namens/ ingrib,outfile,nfiles
+!**********************************************************************
+  read(5,namens) 
+
+  call baopenwt(91,outfile,ierr)  ! open output grib2 file
+        if(ierr.ne.0) then
+          print *,'error writing file ',outfile
+          call abort
+        endif
```

Comment:
Please left indent the whole file.  In fortran 90, there is no need to start at a particular column. Use indents as required to improve code readability and be consistent.